### ISSUE TITLE
Edit Controls can prevent further Arrow Movement between Controls

### ISSUE THREAD

**Issue by jmwmusic** [musescore/MuseScore#20515]
### Issue type

UX/Interaction bug (incorrect behaviour)

### Bug description

In some dialogs and/or toolbars, currently Arrow keys are used to navigate between the controls. Note issue that in dialogs, all controls should be available by pressing Tab(#20056).

If there is an edit control in the set of Arrowable controls, the focus may get stuck at the edit control and it is therefore not clear if some controls cannot be navigated to by keyboard.

### Steps to reproduce

For example:
1. From the Edit menu choose Preferences.
Preferences dialog appears.
2. Select Canvas tab (there are several other examples).
3. Tab.
Focus is on Default Zoom combo box.
4. Arrow down.
Focus is now on mouse zoom precision.
5. Continue to Arrow Up/Down and instead of navigating, now the value in the edit box changes.

It is not possible to know if there are any further controls after the edit box.

Note: this problem cannot be fixed until all controls in dialogs can be focused with Tab.

### MuseScore Version

4.1

### Regression

I don't know

### Operating system

Windows 10 with NVDA

**Comment by ComposingPlanarian** [comment]
True of MuseScore 4.2 too.

I would consider this an accessibility bug.

**Comment by shubham-shinde-442** [comment]
I came across this issue while working on #18880. 

What would be the ideal solution for this issue?

**Comment by cbjeukendrup** [comment]
@shoogle Could you please comment on this?

**Comment by shoogle** [comment]
@shubham-shinde-442, the ideal solution is to fix https://github.com/musescore/MuseScore/issues/20056 so that `Tab` is available for navigation. This would free up the arrow keys to increment the value of a spinner.

**Comment by shoogle** [comment]
@shubham-shinde-442, the ideal solution is to fix #20056 so that `Tab` is used for navigation between neighbouring controls. This would free up the arrow keys to increment the value of a spinner.

<img width="200px" src="https://github.com/user-attachments/assets/31d975c8-57cf-43f3-ad1b-42999aaee604"/>

However, we're not ready to fix #20056 just yet, so in the meantime it's best to implement the same workaround we use in other places, such as text fields in the New Score dialog and number spinners in the Properties panel:

1. Use arrow keys to interact with the spinner or text field control (e.g. to change value, move cursor, etc.).
2. Press Esc to stop iterating with the control.
3. Now you can use arrow keys move away from the control (or press Esc a second time if you want to close the dialog).

**Comment by cbjeukendrup** [comment]
When you get into those controls where you can't move further, then pressing Esc _once_ followed by an arrow key should move to the next control (and there you have to press Esc again, and then arrow again, etc.)

**Comment by jmwmusic** [comment]
Yes you are correct it is possible to get out of it by pressing Esc once.  I have just retested and can see now that the issue is that there are 5 of these in a row for minute, second, milisecond, bar and beat.  So when I pressed escape in most cases pressing an arrow again just got me stuck back in the same stuck position of all 4 arrows saying blank which is why  I didn't realise you can get out of it with Esc.  Without fairly detailed knowledge of the exact layout of this panel memorised navigating it without getting stuck is very confusing & difficult.  Is it not possible for some areas like this to be made tabable before the whole is ready and arrows activate these spin buttons?  From a user point of view consistent uniformity of how the programme behaves is irrelevant if that spot is actually broken and can't be used with keyboard.
